docs/embedded/tutorials/migrate-abs-to-spe.md

```diff
@@ -0,0 +1,328 @@
+        _countdown.Signal();
+
+        return;
+    }
+```
+
+### TransferBlobToSharePointAsync
+```C#
+    Stream blobStream = await _abm.DownloadBlobStreamAsync(blobName);
+
+    string fileName = Path.GetFileName(blobName);
+    await _gcm.UploadStreamToSharePointAsync(blobStream, parentFolderId, fileName);
+```
+
+### DownloadBlobStreamAsync
+```C#
+    BlobClient blobClient = _containerClient.GetBlobClient(blobName);
+
+    MemoryStream memoryStream = new MemoryStream();
+    await blobClient.DownloadToAsync(memoryStream);
+    memoryStream.Position = 0; // Reset the stream position to the beginning
+```
+
+### UploadStreamToSharePointAsync
+```C#
+    int _maxChunkSize = 320 * 1024;
+
+    var uploadSessionRequestBody = new CreateUploadSessionPostRequestBody()
+    {
+        AdditionalData = new Dictionary<string, object>
+        {
+            // Fail is set here, so it doesn't get upload again if it already exist
+            { "@microsoft.graph.conflictBehavior", "fail" }
+        }
+    };
+
+    var uploadSession = await _graphClient.Drives[_containerId]
+        .Items[parentFolderId]
+        .ItemWithPath(fileName)
+        .CreateUploadSession
+        .PostAsync(uploadSessionRequestBody);
+
+    // The stream is the same stream from the downloading the blob
+    var fileUploadTask = new LargeFileUploadTask<DriveItem>(uploadSession, stream, _maxChunkSize, _graphClient.RequestAdapter);
+        IProgress<long> progress = new Progress<long>(prog => Console.WriteLine($"Uploaded {fileName} {prog} bytes"));
+
+    // Check uploadResult.UploadSucceeded to see if it is successful
+    var uploadResult = await fileUploadTask.UploadAsync(progress);
+```
+
+## Handling Errors and Exceptions
+
+### Common Issues
+
+1. File already exists in destination
+    - This app checks to see if the file name exists in the destination before it uploads. If there is a file with the exact same name, it will not do the upload again. It will print to stdout a message that the file already exists. To fix it you can either delete the file from the destination or change the conflictBehavior to replace and not call `CheckIfItemExists` on upload.
+1. Passing a blob list
+    - The blob list is string that would be converted into Json value. The double quotes that wrap each item in the array would need to be escaped. The app checks if this value is in correct format, if it isn't it will print an error to stdout and exit the program.
+    - For example:
+        - Correct: `"[\""example1.txt\", \""example2.txt\", \""example3.txt\"]"`
+        - Incorrect: `"["example1.txt", "example2.txt", "example3.txt"]"`
+1. Not giving enough permission to access the ABS container
+    - The minimum permissions are Read and List
+1. Assigning the correct permissions to the SPE container
+    - The container exists because you can get lists of all the containers. However, when you try to do something to act on the container, it just exits the program.
```

Comment:
I dont understand what this item means. Does it mean assign permission to SPE container exits the sample app but all is actually good?